### PR TITLE
Feature cpp version

### PR DESCRIPTION
**Pull request by hiroaki-ishikawa-t4**
C++ version. Configurable by both rosparam or dynamic reconfigure.

# Difference between python version

* Start sending TF after rosparam or dynamic reconfigure is set (It doesn't send empty value).
* Delete rosparam created by dynamic reconfigure if valid parameter is not set.
    * If rosparam was given before it starts, it keeps it.
    * If rosparam was not given before it starts and the value is given by dynamic reconfigure after run, it will keep created rosparam.
    * If rosparam was not given before run and also dynamic reconfigure is not called, it deletes created (empty) rosparam.
* Remove min/max limitation of tf_x, tf_y, tf_z.

* The C++ version is lacking a function that read parameters directly from yaml specified by rosparam. Please use rosparam function to load yaml.

### CODE REVIEW

**Review by yn-mrse — COMMENTED** [review]
Good job! Thank you for your contribution!
Please correct the following.

**Review by yn-mrse — COMMENTED** [review]
Check and modify the code using roslint.

**Review by yn-mrse — COMMENTED** [review]
If there is a difference between the c ++ specification and the existing python specification, please specify on the PR top page.
In particular, if there is a newly incorporated logic, please give a detailed explanation.

File: include/tunable_static_tf_broadcaster.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    public:
+        TunableStaticTfBroadcaster();
+        ~TunableStaticTfBroadcaster();
+        void run();
+
+    private:
+        ros::NodeHandle private_nh_;
+        tf2_ros::TransformBroadcaster broadcaster_;
+        geometry_msgs::TransformStamped transform_;
+        std::shared_ptr<dynamic_reconfigure::Server<TfConfig> > dynamic_reconfigure_server_;
```
Comment (yn-mrse) [inline include/tunable_static_tf_broadcaster.h]: In ubuntu16.04, ros kinetic environment, an error occurred when building as it is.
Please describe options of c++11 or higher in CMakeList.

File: src/tunable_static_tf_broadcaster.cpp [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    }
+    else
+    {
+        has_initial_static_params_ = false;
+    }
+
+    // Setup dynamic reconfigure
+    dynamic_reconfigure_server_ = std::make_shared<dynamic_reconfigure::Server<TfConfig> >();
+    dynamic_reconfigure::Server<TfConfig>::CallbackType f;
+    f = boost::bind<void>(&TunableStaticTfBroadcaster::dynamicReconfigureCallback, this, _1, _2);
```
Comment (yn-mrse) [inline src/tunable_static_tf_broadcaster.cpp]: In ubuntu16.04, ros kinetic environment, an error occurred when building as it is.
Please remove <void> if there is no problem in other environments.

**Comment by yn-mrse** [comment]
@hiroaki-ishikawa-t4 
The header file is supposed to be deleted, probably because you did not `git add` the header file that changed the path.

**Comment by hiroaki-ishikawa-t4** [comment]
> Check and modify the code using roslint.

Okay I did that.

**Comment by hiroaki-ishikawa-t4** [comment]
> @hiroaki-ishikawa-t4
> The header file is supposed to be deleted, probably because you did not `git add` the header file that changed the path.

I fixed that.

File: include/tunable_static_tf_broadcaster.h (hunk 1 above)
Comment (hiroaki-ishikawa-t4) [reply include/tunable_static_tf_broadcaster.h]: I fixed that.

File: src/tunable_static_tf_broadcaster.cpp (hunk 2 above)
Comment (hiroaki-ishikawa-t4) [reply src/tunable_static_tf_broadcaster.cpp]: I fixed that.

**Review by taichiH — APPROVED** [review]

**Pull request merged**